diff --git a/cli/cppcheckexecutor.cpp b/cli/cppcheckexecutor.cpp
index 2fb4c1380..b78b942a8 100644
--- a/cli/cppcheckexecutor.cpp
+++ b/cli/cppcheckexecutor.cpp
@@ -32,7 +32,7 @@
 #include <algorithm>
 #include <climits>
 
-#if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__CYGWIN__)
+#if defined(__GNUC__) && defined(__GLIBC__) && !defined(__MINGW32__) && !defined(__CYGWIN__)
 #define USE_UNIX_SIGNAL_HANDLING
 #include <execinfo.h>
 #include <cxxabi.h>
@@ -220,7 +220,7 @@ size_t GetArrayLength(const T(&)[size])
  */
 static void print_stacktrace(FILE* f, bool demangling)
 {
-#if defined(__GNUC__)
+#if defined(__GNUC__) && defined(__GLIBC__)
     void *array[32]= {0}; // the less resources the better...
     const int depth = backtrace(array, (int)GetArrayLength(array));
     char **symbolstrings = backtrace_symbols(array, depth);
