diff --git a/lib/mathlib.cpp b/lib/mathlib.cpp
index 16812309f..06f3db9d8 100644
--- a/lib/mathlib.cpp
+++ b/lib/mathlib.cpp
@@ -36,13 +36,13 @@ long MathLib::toLongNumber(const std::string &str)
         || strncmp(str.c_str(), "+0x", 3) == 0
         || strncmp(str.c_str(), "-0x", 3) == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 16);
+        return std::strtoul(str.c_str(), nullptr, 16);
     }
     if (strncmp(str.c_str(), "0" , 1) == 0
         ||  strncmp(str.c_str(), "+0", 2) == 0
         ||  strncmp(str.c_str(), "-0", 2) == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 8);
+        return std::strtoul(str.c_str(), nullptr, 8);
     }
     return (str.find("E", 0) != std::string::npos || str.find("e", 0) != std::string::npos)
            ? static_cast<long>(std::atof(str.c_str()))
@@ -53,7 +53,7 @@ double MathLib::toDoubleNumber(const std::string &str)
 {
     if (strncmp(str.c_str(), "0x", 2) == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 16);
+        return std::strtoul(str.c_str(), nullptr, 16);
     }
     std::istringstream istr(str.c_str());
     double ret;
