diff --git a/lib/mathlib.cpp b/lib/mathlib.cpp
index fe1c16b00..d3cf6918e 100644
--- a/lib/mathlib.cpp
+++ b/lib/mathlib.cpp
@@ -36,13 +36,13 @@ long MathLib::toLongNumber(const std::string &str)
         || str.compare(0, 3, "+0x") == 0
         || str.compare(0, 3, "-0x") == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 16);
+        return std::strtoul(str.c_str(), nullptr, 16);
     }
     if (str.compare(0, 1, "0") == 0
         ||  str.compare(0, 2, "+0") == 0
         ||  str.compare(0, 2, "-0") == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 8);
+        return std::strtoul(str.c_str(), nullptr, 8);
     }
     return (str.find("E", 0) != std::string::npos || str.find("e", 0) != std::string::npos)
            ? static_cast<long>(std::atof(str.c_str()))
@@ -53,7 +53,7 @@ double MathLib::toDoubleNumber(const std::string &str)
 {
     if (str.compare(0, 2, "0x") == 0)
     {
-        return std::strtoul(str.c_str(), '\0', 16);
+        return std::strtoul(str.c_str(), nullptr, 16);
     }
     // nullcheck
     else if (isNullValue(str))
